mithril-common/src/test_utils/certificate_chain_builder.rs

```diff
@@ -760,32 +789,66 @@ mod test {
 
     #[test]
     fn builds_certificate_chain_correctly_chained() {
-        let certificates = vec![
-            Certificate {
-                epoch: Epoch(1),
-                ..fake_data::certificate("cert-1".to_string())
-            },
-            Certificate {
-                epoch: Epoch(2),
-                ..fake_data::certificate("cert-2".to_string())
-            },
+        fn create_fake_certificate(epoch: Epoch, index: u64) -> Certificate {
             Certificate {
-                epoch: Epoch(3),
-                ..fake_data::certificate("cert-3".to_string())
-            },
+                epoch,
+                signed_message: format!("certificate-{index}"),
+                ..fake_data::certificate("cert-fake".to_string())
+            }
+        }
+
+        let certificate_1 = create_fake_certificate(Epoch(1), 1);
+        let certificate_2 = create_fake_certificate(Epoch(2), 2);
+        let certificate_3 = create_fake_certificate(Epoch(2), 3);
+        let certificate_4 = create_fake_certificate(Epoch(3), 4);
+        let certificate_5 = create_fake_certificate(Epoch(4), 5);
+        let certificate_6 = create_fake_certificate(Epoch(4), 6);
+        let certificate_7 = create_fake_certificate(Epoch(4), 7);
+        let certificates = vec![
+            certificate_1,
+            certificate_2,
+            certificate_3,
+            certificate_4,
+            certificate_5,
+            certificate_6,
+            certificate_7,
         ];
 
-        let certificates_chained =
+        let mut certificates_chained =
             CertificateChainBuilder::default().compute_chained_certificates(certificates);
+        certificates_chained.reverse();
 
-        assert_eq!("", certificates_chained[2].previous_hash);
+        let certificate_chained_1 = &certificates_chained[0];
+        let certificate_chained_2 = &certificates_chained[1];
+        let certificate_chained_3 = &certificates_chained[2];
+        let certificate_chained_4 = &certificates_chained[3];
+        let certificate_chained_5 = &certificates_chained[4];
+        let certificate_chained_6 = &certificates_chained[5];
+        let certificate_chained_7 = &certificates_chained[6];
+        assert_eq!("", certificate_chained_1.previous_hash);
+        assert_eq!(
+            certificate_chained_2.previous_hash,
+            certificate_chained_1.hash
+        );
+        assert_eq!(
+            certificate_chained_3.previous_hash,
+            certificate_chained_2.hash
+        );
+        assert_eq!(
+            certificate_chained_4.previous_hash,
+            certificate_chained_2.hash
```

Comment:
I don't understand why we compare previous_hash from 4th with hash from 2nd (idem with 7th and 5th).
If it's an error in test, there is  probably because all hashes are equals and it's a problem to really test values.